### PR TITLE
Indexing on Location fields

### PR DESCRIPTION
**Pull request by KhadijaMahanga**
- Add indexes on country
- Add indexes on city field

### CODE REVIEW

File: feinstaub/sensors/models.py [hunk 1]
```diff
@@ -175,6 +175,10 @@ class SensorLocation(TimeStampedModel):
 
     class Meta:
         ordering = ['location', ]
+        indexes = [
+            models.Index(fields=['country', 'city']),
+            models.Index(fields=['city'], name='city_idx'),
```
Comment (kilemensi) [inline feinstaub/sensors/models.py:180]: Any difference with adding [`db_index`](https://docs.djangoproject.com/en/2.2/ref/models/fields/#db-index) on the field itself?

**Review by kilemensi — COMMENTED** [review]
👍🏾 @KhadijaMahanga ... On mobile so can’t run it:

1. No migration file generated? Or are we keeping them somewhere else?
2. Any reason we need index on country, city and country & city combined?

**Comment by KhadijaMahanga** [comment]
@kilemensi  
1. I planned to have the migration file on `sensors.AFRICA-api`. But I guess having it here makes more sense. Let me run it and add it here
2. I added `city & country` for the case where we want to filter on both city & county. My assumption is we may have similar city names for different country

**Comment by kilemensi** [comment]
> @kilemensi
> 
> 1. I planned to have the migration file on `sensors.AFRICA-api`. But I guess having it here makes more sense. Let me run it and add it here
> 2. I added `city & country` for the case where we want to filter on both city & county. My assumption is we may have similar city names for different country

1. If that’s how we’ve been doing it, lets keep it that way.
2. If they both have indexes I assume DB will be smart enough to use them rather than us creating another index for just this special case.

**Comment by KhadijaMahanga** [comment]
@kilemensi alright

**Comment by KhadijaMahanga** [comment]
@kilemensi let me put the migration files here. So even if one need to use this repo as standalone, the db isn't messed up. Or once we are ready to push to `opendata-stuttgart`

File: feinstaub/sensors/models.py (hunk 1 above)
Comment (KhadijaMahanga) [reply feinstaub/sensors/models.py:180]: @kilemensi I tested this, and django did not update the migration file to create an index. Apparently, it creates if the table is nonexistent and first created.

**Comment by KhadijaMahanga** [comment]
@kilemensi @esirK 
The migration file here includes these two files on sensor api [1](https://github.com/CodeForAfrica/sensors.AFRICA-api/blob/master/sensorsafrica/openstuttgart/feinstaub/sensors/migrations/0020_auto_20190314_1232.py) and [2](https://github.com/CodeForAfrica/sensors.AFRICA-api/blob/master/sensorsafrica/openstuttgart/feinstaub/sensors/migrations/0021_auto_20210204_1106.py)

**Comment by kilemensi** [comment]
> @kilemensi @esirK
> The migration file here includes these two files on sensor api [1](https://github.com/CodeForAfrica/sensors.AFRICA-api/blob/master/sensorsafrica/openstuttgart/feinstaub/sensors/migrations/0020_auto_20190314_1232.py) and [2](https://github.com/CodeForAfrica/sensors.AFRICA-api/blob/master/sensorsafrica/openstuttgart/feinstaub/sensors/migrations/0021_auto_20210204_1106.py)

Totally lost now but I think the big question is: are we planning on merging back to feinstaub-api? If the answer is yes, then my understanding is they no longer keep individual migration files in the repo (or at least that's why Karim started keeping the migrations in `sensors.africa-api`).

**Comment by KhadijaMahanga** [comment]
@kilemensi alright I didn't know that was the case. We will stick to having the migration files on `sensors.AFRICA-api`

File: feinstaub/sensors/models.py [hunk 2]
```diff
@@ -175,6 +175,10 @@ class SensorLocation(TimeStampedModel):
 
     class Meta:
         ordering = ['location', ]
+        indexes = [
+            models.Index(fields=['country'], name='country_idx'),
```
Comment (kilemensi) [inline feinstaub/sensors/models.py:179]: Doesn't Django generate name for these?

**Review by kilemensi — APPROVED** [review]
🚀

**Review by esirK — APPROVED** [review]

File: feinstaub/sensors/models.py (hunk 2 above)
Comment (KhadijaMahanga) [reply feinstaub/sensors/models.py:179]: yes it does, but you can also specify @kilemensi

**Pull request merged**